dask_geomodeling/raster/elemwise.py

```diff
@@ -746,3 +749,81 @@ def process(process_kwargs, *args):
             values[index] = data[index]
 
         return {"values": values, "no_data_value": fillvalue}
+
+
+class BaseLogExp(BaseSingle):
+    """
+    Base Block for Exp, Log and Log10 Blocks.
+    """
+    def __init__(self, x):
+        if x.dtype == np.dtype("bool"):
```

Comment:
I think it is good to warn the users if the input is boolean dtype. It makes no sense to take Log or Exp from that.

Still, numpy just interprets it as 0 and 1 and nicely gives you back some numbers. What were your deliberations on this check?

Reply:
I know numpy doesn't complain, but dask_geomodeling complaining might stop someone from doing something unintended.